File: StudentMentalHealthCounselingSystem/frontend/src/types/appointment.ts
import type { User } from './user';

export type AppointmentStatus = 'PENDING' | 'CONFIRMED' | 'COMPLETED' | 'CANCELLED';

export interface Appointment {
  id: number;
  student: User;
  counselor: User;
  subject: string;
  description: string;
  appointmentTime: string;
  status: AppointmentStatus;
  createTime: string;
  updateTime: string;
}

export interface AppointmentListRequest {
  page: number;
  size: number;
  status?: AppointmentStatus;
  statuses?: AppointmentStatus[];
  startTime?: string;
  endTime?: string;
}

export interface AppointmentListResponse {
  content: Appointment[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}

export interface CreateAppointmentRequest {
  counselorId: number;
  subject: string;
  description: string;
  appointmentTime: string;
} 

File: StudentMentalHealthCounselingSystem/backend/src/main/java/com/mentalhealth/counseling/entity/Appointment.java
package com.mentalhealth.counseling.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * 预约实体类
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "appointments")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private User student;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "counselor_id", nullable = false)
    private User counselor;

    @Column(nullable = false)
    private String subject;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private LocalDateTime appointmentTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AppointmentStatus status = AppointmentStatus.PENDING;

    @CreationTimestamp
    private LocalDateTime createTime;

    @UpdateTimestamp
    private LocalDateTime updateTime;

    public enum AppointmentStatus {
        PENDING,    // 待确认
        CONFIRMED,  // 已确认
        COMPLETED,  // 已完成
        CANCELLED   // 已取消
    }
} 